src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -238,14 +238,18 @@ public uint GetConnectionId(SNIHandle handle, ref Guid clientConnectionId)
         /// <returns>SNI error status</returns>
         public uint WritePacket(SNIHandle handle, SNIPacket packet, bool sync)
         {
+            SNIPacket clonedPacket = packet.Clone();
```

Comment:
This can be changed to 
``` C#
using(SNIPacket clonedPacket = packet.Clone()) {
    // send logic
}
```

This way even if an exception occurs in Send or SendAsync will result in clonedPacket being disposed

Reply:
On Second thoughts, should the Cloned packet be disposed here? SendAsync will return immideately, which means that the I/O may not complete and the Dispose will set the _data to null and allow it to be GCed before I/O completes.

Reply:
fixed.